test/reference_tests.jl

```diff
@@ -0,0 +1,71 @@
+connections = [
+    (1, 2, 1100),
+    (2, 4, 300),
+    (6, 2, 1400),
+    (2, 3, 500),
+    (2, 5, 300),
+    (5, 7, 100),
+    (2, 8, 100),
+    (3, 9, 150),
+    (2, 10, 500),
+    (10, 11, 50),
+    (10, 12, 80),
+    (5, 13, 150),
+    (3, 14, 100),
+    (10, 15, 300),
+]
+
+labels = [
+    "Salary",
+    "Income",
+    "Rent",
+    "Insurance",
+    "Car",
+    "Salary 2",
+    "Depreciation",
+    "Internet",
+    "Electricity",
+    "Food & Household",
+    "Fast Food",
+    "Drinks",
+    "Gas",
+    "Water",
+    "Groceries",
+]
+
+Random.seed!(41)
```

Comment:
Please don't use a global random seed, that makes the tests order-dependent. In fact, I prefer not using random values at all, usually it's best to craft some values yourself for which you can tell if they've been applied correctly. If things are random, we don't notice weird patterns quite as well.

For example, you could do `nodecolor = Makie.to_colormap(:tab20)[1:n]`